client/src/App.jsx

```diff
@@ -1,15 +1,168 @@
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+
+  // Job Updates State
+  const [jobUpdates] = useState(mockJobUpdates);
+
+  // Admin State
+  const [managedUsers, setManagedUsers] = useState(mockManagedUsers);
+
+  // Reset all states (when user logs out)
+  const resetState = () => {
+    setCurrUser({
+      userID: "",
+      firstName: "",
+      lastName: "",
+      email: "",
+      isAdmin: false,
+    });
+    setIsAuthenticated(false);
+    setManagedUsers([]);
+  };
+
+  // Setting state (when user logs in)
+  const loginUser = (userData) => {
+    setCurrUser({
+      userID: userData.userID || userData.user_id || 1,
+      firstName: userData.firstName || userData.first_name || "",
+      lastName: userData.lastName || userData.last_name || "",
+      email: userData.email || userData.last_name,
+      isAdmin: userData.isAdmin || userData.is_admin || false,
+    });
+    setIsAuthenticated(true);
+  };
+
+  // declaring routes here
   return (
-    <div className="App">
-      <DashboardPage />
-    </div>
+    <Router>
+      <Routes>
+        <Route
+          path="/"
+          element={
+            isAuthenticated ? (
+              <Navigate to="/dashboard" />
+            ) : (
+              <Navigate to="/signin" />
+            )
+          }
+        />
+        <Route
+          path="/signin"
+          element={
+            <RouteGuard
+              isPermitted={!isAuthenticated}
+              redirect={dashboardRedirect}
+            >
+              <LoginPage
+                setIsAuthenticated={setIsAuthenticated}
+                loginUser={loginUser}
+              />
+            </RouteGuard>
+          }
+        />
+        <Route path="/logout" element={<LogoutPage onLogout={resetState} />} />
+        <Route
+          path="/dashboard"
+          element={
+            <AuthGuard isAuthenticated={isAuthenticated} loginUser={loginUser}>
+              <DashboardPage currUser={currUser} jobUpdates={jobUpdates} />
+            </AuthGuard>
+          }
+        />
+        <Route
+          path="/admin"
+          element={
+            <AuthGuard isAuthenticated={isAuthenticated} loginUser={loginUser}>
+              <RouteGuard
+                isPermitted={currUser.isAdmin}
+                redirect={dashboardRedirect}
+              >
+                <ManagedUsersLoader
+                  currUser={currUser}
+                  setManagedUsers={setManagedUsers}
+                >
+                  <AdminDashboard
+                    managedUsers={managedUsers}
+                    setManagedUsers={setManagedUsers}
+                  />
+                </ManagedUsersLoader>
+              </RouteGuard>
+            </AuthGuard>
+          }
+        />
+        <Route
+          path="/admin/create-user"
+          element={
+            <AuthGuard
+              isAuthenticated={isAuthenticated}
+              loginUser={loginUser}
+              redirectUrl={adminRedirect}
```

Comment:
Yeah, so if we are on create-user or edit-user i was thinking if someone refresh or hard url navigates to this url, it would be best to navigate gate them to the admin dashboard instead of letting them go straight to edit user and create user.

Reply:
Ok that sounds good, we can leave it as it is.